omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java

```diff
@@ -910,7 +894,65 @@ private void addPaths() {
 		Collections.sort(tagList);
 		swaggerSpecification.setTags(tagList);
 	}
-	
+
+	/**
+	 * Extracts resource name from a resource handler
+	 * @param trim if false returns resource name with version, if true version part of the name is trimmed
+	 * @return resource name
+	 */
+	private String getResourceName(DelegatingResourceHandler resourceHandler, boolean trim) {
+		Resource annotation = resourceHandler.getClass().getAnnotation(Resource.class);
+
+		String resourceName = null;
+		if (annotation != null) {
+			// top level resource
+			resourceName = annotation.name();
+		} else {
+			// sub resource
+			SubResource subResourceAnnotation = resourceHandler.getClass().getAnnotation(SubResource.class);
+			if (subResourceAnnotation != null) {
+				resourceName = subResourceAnnotation.path();
+			}
+		}
+		if (trim) {
+			resourceName = trimResourceName(resourceName);
+		}
+		return resourceName;
+	}
+
+	/**
+	 * Extracts resource parent name from a resource handler
+	 * @param trim if false returns name with version, if true version part of the name is trimmed
+	 * @return name of the parent resource or null if its not a sub-resource
+	 */
+	private String getResourceParentName(DelegatingResourceHandler resourceHandler, boolean trim) {
+		SubResource subResourceAnnotation = resourceHandler.getClass().getAnnotation(SubResource.class);
```

Comment:
Think about it this way, by the time one calls `getResourceParentName()`, they already know that the resource handler is a sub resource, otherwise how do they determine which one of your 2 methods to call in your current code? In fact, I kind of agree with you that client code should not have to know if they are dealing with a resource vs sub resource, this makes me think that you should actually have one method and in your code you should check if resource handler is a resource vs a sub resource via reflection